code/game/machinery/doors/firedoor.dm

```diff
@@ -142,6 +142,9 @@
 	if(density)
 		being_held_open = TRUE
 		open()
+		if(QDELETED(user))
+			being_held_open = FALSE
+			return
 		RegisterSignal(user, COMSIG_MOVABLE_MOVED, .proc/handle_held_open_adjacency)
 		RegisterSignal(user, COMSIG_LIVING_SET_BODY_POSITION, .proc/handle_held_open_adjacency)
```

Comment:
While the QDELETED check is good, I meant you should also register for a signal if `user` gets qdel'd after as well.

Reply:
let me know if this is enough. I removed the QDELETED check from earlier to keep it tidier and easier to read, since it's nigh impossible for this proc to trigger on qdeleted mobs, I think.

Reply:
I'm pretty sure `open()` sleeps actually, so the mob could theoretically get dusted while it's happening. Can you bring it back?

Reply:
Copy that, I'll readd it.